src/index.ts

```diff
@@ -0,0 +1,313 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) 2014-2015, PhosphorJS Contributors
+|
+| Distributed under the terms of the BSD 3-Clause License.
+|
+| The full license is in the file LICENSE, distributed with this software.
+|----------------------------------------------------------------------------*/
+'use strict';
+
+import {
+  IDisposable
+} from 'phosphor-disposable';
+
+import {
+  Message
+} from 'phosphor-messaging';
+
+import {
+  Signal, ISignal, clearSignalData
+} from 'phosphor-signaling';
+
+import {
+  Widget, ResizeMessage
+} from 'phosphor-widget';
+
+
+/**
+ * The interface required for an editor model.
+ */
+export
+interface IEditorModel {
+  /**
+   * Saves the current data in the existing file.
+   *
+   * @param data - A string containing the complete data to be saved.
+   */
+  save(data: string): void;
+  /**
+   * Renames the current file.
+   *
+   * @param name - A string denoting the new filename.
+   *
+   * #### Notes
+   * This does *not* remove the existing file, it simply saves a
+   * new file with the same data.
+   */
+  rename(name: string): void;
+}
+
+
+/**
+ * An implementation of IEditorModel.
+ */
+export
+class EditorModel implements IEditorModel {
+
+  /**
+   * Save the data to file.
+   */
+  save(data: string): void {
+    // to be delegated to jupyter-js-services.
+    console.log('model save');
+  }
+
+  /**
+   * Rename the current file.
+   */
+  rename(name: string): void {
+    // to be delegated to jupyter-js-services.
+    console.log('model rename');
+  }
+
+  private _filename: string;
+  private _mode: string;
+  private _lineNumbers = true;
+  private _tabSize = 2;
+}
+
+
+/**
+ * An interface required for implementing the view-model
+ * for an editor.
+ */
+export
+interface IEditorViewModel {
+  /**
+   * Save the current text buffer to the file.
+   */
+  save(data: string): void;
+  /**
+   * Rename this file.
+   */
+  rename(name: string): void;
+  /**
+   * The mode of the editor, eg. for code this would be
+   * the language, like 'python' or 'javascript'.
+   */
+  mode(): string;
+  /**
+   * Sets the view class on on the view model.
+   */
+  setView(view: IEditorWidget): void;
```

Comment:
That's not the responsibility of the view model. That connection should be handled in the view, and converted to a method call on the view model. i.e. move the timer to the widget.